From 4ecb79d8980fcf7cd0ff25c8c21bf473156fdb19 Mon Sep 17 00:00:00 2001
From: David Seifert <soap@gentoo.org>
Date: Tue, 15 Feb 2022 11:24:54 +0100
Subject: [PATCH] Build uhd_test library as static when
 `-DBUILD_SHARED_LIBS=ON`

* A uhd_test.so lacks the necessary symbols for testing
  due to default visibility set to hidden.
---
 tests/common/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/common/CMakeLists.txt b/tests/common/CMakeLists.txt
index ac0e1b4492..36c9f9b076 100644
--- a/tests/common/CMakeLists.txt
+++ b/tests/common/CMakeLists.txt
@@ -8,6 +8,6 @@
 # Build uhd_test static lib
 ########################################################################
 include_directories("${UHD_SOURCE_DIR}/lib/include")
-add_library(uhd_test
+add_library(uhd_test STATIC
     ${CMAKE_CURRENT_SOURCE_DIR}/mock_zero_copy.cpp
 )
